RecoBTag/SecondaryVertex/src/CandidateBoostedDoubleSecondaryVertexComputer.cc

```diff
@@ -13,39 +20,73 @@ CandidateBoostedDoubleSecondaryVertexComputer::CandidateBoostedDoubleSecondaryVe
+                                      "tau_vertexMass_0","tau_vertexEnergyRatio_0","tau_vertexDeltaR_0","tau_flightDistance2dSig_0",
+                                      "tau_vertexMass_1","tau_vertexEnergyRatio_1","tau_flightDistance2dSig_1",
+                                      "jetNTracks","nSV"});
+  // book TMVA readers
+  std::vector<std::string> spectators({"massPruned", "flavour", "nbHadrons", "ptPruned", "etaPruned"});
+
+  if (useCondDB_)
+  {
+     const GBRWrapperRcd & gbrWrapperRecord = record.getRecord<GBRWrapperRcd>();
 
-  mvaID->initialize("Color:Silent:Error", "BDTG", weightFile_.fullPath(), variables, spectators,true,false);
-}
+     edm::ESHandle<GBRForest> gbrForestHandle;
+     gbrWrapperRecord.get(gbrForestLabel_.c_str(), gbrForestHandle);
 
+     mvaID->initializeGBRForest(gbrForestHandle.product(), variables, spectators, useAdaBoost_);
+  }
+  else
+    mvaID->initialize("Color:Silent:Error", "BDT", weightFile_.fullPath(), variables, spectators, useGBRForest_, useAdaBoost_);
+
+  // get TransientTrackBuilder
+  const TransientTrackRecord & transientTrackRcd = record.getRecord<TransientTrackRecord>();
+  transientTrackRcd.get("TransientTrackBuilder", trackBuilder);
+}
 
 float CandidateBoostedDoubleSecondaryVertexComputer::discriminator(const TagInfoHelper & tagInfo) const
 {
   // get TagInfos
   const reco::CandIPTagInfo              & ipTagInfo = tagInfo.get<reco::CandIPTagInfo>(0);
   const reco::CandSecondaryVertexTagInfo & svTagInfo = tagInfo.get<reco::CandSecondaryVertexTagInfo>(1);
-  const reco::CandSoftLeptonTagInfo      & muonTagInfo = tagInfo.get<reco::CandSoftLeptonTagInfo>(2);
-  const reco::CandSoftLeptonTagInfo      & elecTagInfo = tagInfo.get<reco::CandSoftLeptonTagInfo>(3);
 
   // default discriminator value
   float value = -10.;
 
   // default variable values
-  float z_ratio = -1. , tau_dot = -1., SV_pt_0 = -1., SV_mass_0 = -1., SV_EnergyRatio_0 = -1., SV_EnergyRatio_1 = -1., tau21 = -1.;
-  int contSV = 0, vertexNTracks = 0;
-  int nSL = 0, nSM = 0, nSE = 0;
+  float z_ratio = -3.;
+  float trackSip3dSig_3 = -50., trackSip3dSig_2 = -50., trackSip3dSig_1 = -50., trackSip3dSig_0 = -50.;
+  float tau2_trackSip3dSig_0 = -50., tau1_trackSip3dSig_0 = -50., tau2_trackSip3dSig_1 = -50., tau1_trackSip3dSig_1 = -50.;
+  float trackSip2dSigAboveCharm_0 = -19., trackSip2dSigAboveBottom_0 = -19., trackSip2dSigAboveBottom_1 = -19.;
+  float tau1_trackEtaRel_0 = -1., tau1_trackEtaRel_1 = -1., tau1_trackEtaRel_2 = -1.;
+  float tau2_trackEtaRel_0 = -1., tau2_trackEtaRel_1 = -1., tau2_trackEtaRel_2 = -1.;
+  float tau1_vertexMass = -1., tau1_vertexEnergyRatio = -1., tau1_vertexDeltaR = -1., tau1_flightDistance2dSig = -1.;
+  float tau2_vertexMass = -1., tau2_vertexEnergyRatio = -1., tau2_vertexDeltaR = -1., tau2_flightDistance2dSig = -1.;
+  float jetNTracks = 0, nSV = 0, tau1_nSecondaryVertices = 0, tau2_nSecondaryVertices = 0;
```

Comment:
@ferencek:
I was not suggesting that variables like trackSip3dSig_3 be declared const.  What's needed is to add ` static constexpr float dummySignificance = -50.0f;` and then use it in place of -50 throughout the file.